scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/compile/NodeParametersAdjuster.scala

```diff
@@ -0,0 +1,72 @@
+package pl.touk.nussknacker.engine.compile
+
+import cats.Id
+import cats.data.WriterT
+import cats.implicits.toTraverseOps
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.api.{JobData, NodeId}
+import pl.touk.nussknacker.engine.api.definition.{Parameter => ParameterDefinition}
+import pl.touk.nussknacker.engine.api.parameter.ParameterName
+import pl.touk.nussknacker.engine.compile.CompilationLoggerExtensions.Ops
+import pl.touk.nussknacker.engine.graph.evaluatedparam.{Parameter => NodeParameter}
+import pl.touk.nussknacker.engine.util.Implicits.RichTupleList
+
+object NodeParametersAdjuster extends LazyLogging {
+
+  def adjustNonBranchParameters(parameterDefinitions: List[ParameterDefinition], parameters: List[NodeParameter])(
+      implicit nodeId: NodeId,
+      jobData: JobData
+  ): List[NodeParameter] = {
+    val nonBranchParametersDefinition = parameterDefinitions.filter(!_.branchParam)
+    checkRedundancyAndWarnIfNeeded(nonBranchParametersDefinition, parameters)
+    addMissingNodeParameters(nonBranchParametersDefinition, parameters)
+  }
+
+  private def checkRedundancyAndWarnIfNeeded(
+      parameterDefinitions: List[ParameterDefinition],
+      parameters: List[NodeParameter]
+  )(
+      implicit nodeId: NodeId,
+      jobData: JobData
+  ): Unit = {
+    val redundantParams = parameters.map(_.name).toSet.diff(parameterDefinitions.map(_.name).toSet)
+    if (redundantParams.nonEmpty) {
+      logger.compilationWarning(
+        s"Found redundant parameters: ${redundantParams.toList.map(_.value).sorted.mkString(", ")}. They will be skipped."
+      )
+    }
+  }
+
+  private def addMissingNodeParameters(
+      parameterDefinitions: List[ParameterDefinition],
+      parameters: List[NodeParameter]
+  )(
+      implicit nodeId: NodeId,
+      jobData: JobData
+  ) = {
+    val parameterByName = parameters.map(p => p.name -> p).toMapCheckingDuplicates
+    val (missingParameterNames, adjustedParameters) = parameterDefinitions
```

Comment:
I think we can simplify this by:
1. using simpler Writer or even just `foldLeft` 
2. `traverse` instead of `map ... sequence` 

Like
 ```scala
 val (missingParameterNames, adjustedParameters) =
      parameterDefinitions.traverse { param =>
        parameterByName.get(param.name) match {
          case Some(param) => Writer.value[Set[ParameterName], NodeParameter](param)
          case None =>
            val defaultParam = NodeParameter(param.name, param.finalDefaultValue)
            Writer(Set(param.name), defaultParam)
        }
}.run
```